utils/gitlab_api.py

```diff
@@ -44,7 +44,21 @@ def delete_file(self, branch_name, file_path, commit_message):
                 {
                     'action': 'delete',
                     'file_path': file_path
-                    }
+                }
+            ]
+        }
+        self.project.commits.create(data)
+
+    def update_file(self, branch_name, file_path, commit_message, content):
```

Comment:
perhaps wortth supporting a list of files to update?

`def update_file(self, branch_name, commit_message, file_contents):`

where `file_contents` is a dictionary where the key is the path and the value is the content? or a list of tuples, where the first element is the path, and the second the content?

It's a minor change and increases reusability.

Reply:
each time `update_file` is called, it creates a commit. we are later creating a single MR containing all commits.
i think its better this way. wdyt?

Reply:
if we do it like I proposed we leave it up to the user of this library

Reply:
agreed to be done in future iteration.